src/main/java/org/amahi/anywhere/activity/NavigationActivity.java

```diff
@@ -337,4 +338,25 @@ protected void onPause() {
 
 		BusProvider.getBus().unregister(this);
 	}
+
+    boolean doubleBackToExitPressedOnce = false;
+
+    @Override
+    public void onBackPressed() {
+        if (doubleBackToExitPressedOnce) {
+            super.onBackPressed();
+            return;
+        }
+
+        this.doubleBackToExitPressedOnce = true;
+        Toast.makeText(this, "Please click BACK again to exit", Toast.LENGTH_SHORT).show();
```

Comment:
Please use the xml resource here that you have added in `strings.xml` instead of hardcoding.